Frameworks/Account/Account.swift

```diff
@@ -307,6 +309,28 @@ public final class Account: DisplayNameProvider, UnreadCountProvider, Container,
 			break
 		}
 	}
+	
+	public static func oauthAuthorizationCodeGrantRequest(for type: AccountType, client: OAuthAuthorizationClient) -> URLRequest {
+		switch type {
+		case .feedly:
+			return FeedlyAccountDelegate.oauthAuthorizationCodeGrantRequest(for: client)
+		default:
+			fatalError("\(type) does not support OAuth authorization code granting.")
+		}
```

Comment:
We don't usually use fatalError and use assertionFailure instead.

Reply:
For release style builds the assertion would be stripped and I'd be required to return a value for these cases. Returning an "empty" URL request seemed like a hack so I figured it was reasonable to fatalError.

Reply:
If I read this right: in this case it’s a serious programming error and not something that would normally happen, even in an edge case on a bad day. So I’m okay with a fatalError (or precondition check) for this.

Reply:
That's how I understand the situation, too. The programmer is trying access behaviours of certain objects that by definition they are not capable of. I expect the most basic verification of this functionality during development will have revealed this to prevent it occuring in the wild.